test/benchmark/Benchmark_Driver.test-sh

```diff
@@ -0,0 +1,21 @@
+// REQUIRES: OS=macosx
+// REQUIRES: asserts
+// REQUIRES: CMAKE_GENERATOR=Ninja
+
+// Integration tests between Benchmark_Driver and Benchmark_O
```

Comment:
This is the wrong place for this sort of test. We should be able to run benchmark tests against a standalone build of the benchmarks.

Reply:
Let me be more specific.

* If we /do/ want to put the test here, we need to have a "lit" feature that says that benchmarks are available and if this feature is not set, do not run this test. Otherwise, we will try and run this test when the benchmarks are not enabled and the test will fail. So no matter what, we need that.

* The benchmark's standalone configuration should be able to invoke these tests independently via its cmake. Now I am not saying that this PR is the right place to teach the benchmark's cmake how to do that. Rather I am suggesting that by doing the PR in this way, that becomes more difficult. Instead, we should follow the model of the validation-tests where despite validation-tests not being in the test directory itself, we have taught lit how to delegate to it. The way that this is done can be seen in the file ./test/CMakeLists.txt. I am happy to answer questions/pointers here.

Reply:
I had these tests under `validation-test/benchmark` (next to the `generate-harness.test-sh`) originally, but moved them to `test/benchmark` where they logically belong, as they should be run more often. That being said, looking at `test/CmakeLists.txt` reveals no special handling for `benchmark` anywhere, so I really don't know what you mean.

Reply:
Only delegation we did when introducing validation-tests for `generate-harness.py` was modification of `test/lit.cfg`, where we've added substitution for `%swift_src_root'`. This was also what I had to do now by introducing `%Benchmark_O` and `%Benchmark_Driver` using `inferSwiftBinary` in `test/lit.cfg`.  See [this commit](https://github.com/apple/swift/pull/12415/commits/6afb464aeed10d10a16100bb090e7425e9b2b731#diff-f913ebb5bc611de19290c715fcdad913). Is this what you had in mind?